src/scala/com/github/johnynek/bazel_deps/Writer.scala

```diff
@@ -7,40 +7,48 @@ import scala.util.{ Failure, Success, Try }
 
 object Writer {
 
+  /**
+   * Takes a BUILD file path and generated contents, and returns the formatted version of those contents (e.g. with
+   * buildifier).
+   */
+  type BuildFileFormatter = ((IO.Path, String) => String)
```

Comment:
I think this needs to be `(IO.Path, String) => IO[String]`. We can't run this command until the other IO before has run. I don't see a test here, but I'm concerned this is not happening in the right order.

I think to do this. It seems like currently we are doing a side effect that happens to be writing to the same file that we later write to.

If that's right, I'm not very happy with that approach.

I'm actually not sold on integrating with buildifier in this tool. Can't you just run buildifier afterwards in a script?

Reply:
It doesn't need to be an `IO` because buildifier isn't operating on the BUILD file after it's written. Instead, buildifier is run on the in-memory file contents before they are passed to the IO stack.  Buildifier takes the generated source on stdin and formats to stdout, without touching the filesystem.  It is a side-effect, since it runs a process, but it's not IO.

The driving motivation for doing the formatting here, in bazel-deps and at this point in generation, is to make the `--check-only` flag play well with the buildifier.  I didn't realize until I was trying to use that new flag on one of our big repos that our existing approach -- running the buildifier separately after `generate` -- makes `--check-only` useless.  It's comparing the unformatted, raw generated files with the formatted files that are checked in, so it always finds differences.  By formatting inside bazel-deps, the generated files for both writing and comparing will be formatted identically, and everything matches up.

Reply:
okay. sounds good. I misunderstood.

Let's merge when we get CI green.